test/unit/conftest.py

```diff
@@ -121,3 +136,35 @@ def verify(self, cert):
             return VerificationSuccess()
 
     return NullPolicy()
+
+
+@pytest.fixture
+def mock_staging_tuf(monkeypatch):
+    """Mock that prevents tuf module from making requests: it returns staging
+    assets from a local directory instead
+
+    Return a tuple of dicts with the requested files and counts"""
+
+    success = defaultdict(int)
+    failure = defaultdict(int)
+
+    class MockFetcher(FetcherInterface):
+        def _fetch(self, url: str):
+            filename = os.path.basename(url)
+            filepath = _TUF_ASSETS / filename
+            if filepath.is_file():
+                success[filename] += 1
+                return open(filepath, "rb")
```

Comment:
I suppose this should actually return a function that yields the content, otherwise the test leaves the file unclosed

Reply:
That's worth a `NOTE` or `TODO` in the source IMO, but not a big deal otherwise -- I'm okay with cleanup at process destruction time for unit tests 🙂